dicegame.js

```diff
@@ -1,11 +1,40 @@
-const dice1 = [1,2,3,4,5,6];
-const dice2 = [1,2,3,4,5,6];
+// what are the differences and advantages of let vs const or var?
+// how many functions have you used? How many could you use?
+// how would you structure this as one or more functions?
+// what is the difference between Functional, Object-Oriented and Procedural Programming?
+// how could you make this DRY and adaptable to different numbers of dice and dice types?
 
-console.log("rolling dice");
+const generateDice = (sides) => {
+  const number = Math.floor(Math.random() * sides + 1)
+  return number;
+}
 
-let dice1rand = dice1[Math.floor(Math.random() * dice1.length)];
-let dice2rand = dice2[Math.floor(Math.random() * dice2.length)];
-let diceSum = dice1rand + dice2rand;
+const diceAmount = (sides,number) => {
+  const dice = [];
+  for (die = 0; die < number; die++) {
+    dice.push(generateDice(sides));
+  }
+  return dice;
+}
+
+const playGame = (sides, number) => {
+  const diceArray = diceAmount(sides, number);
+  const diceSum = diceArray.reduce((accumulator, currentValue) => accumulator + currentValue);
+  return diceSum;
+}
+
+const myArgs = process.argv.slice(2);
+
+const result = () => {
+  const resultSum = playGame(myArgs[0], myArgs[1]);
+  return console.log(resultSum);
+};
+
+if (result() === 10) {
```

Comment:
why 10?

Reply:
Secondly, think about what the return type is from `result()`... what are we actually comparing?

Reply:
It looks like this will always return `false` 

What are all the possible return values of `result()`? What do you want them to be?

We should separate out side-effects as much as possible. 
Logging is a side-effect 😄 
Do we ever want to return a side-effect? 🤔